src/main/java/io/stargate/sgv2/jsonapi/service/projection/ProjectionLayer.java

```diff
@@ -89,49 +81,51 @@ private static ProjectionLayer buildLayers(
           failOnOverlap,
           DocumentConstants.Fields.DOC_ID,
           root,
-          new String[] {DocumentConstants.Fields.DOC_ID});
+          ProjectionPath.from(DocumentConstants.Fields.DOC_ID));
     }
     if (add$vector) {
       buildPath(
           failOnOverlap,
           DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD,
           root,
-          new String[] {DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD});
+          ProjectionPath.from(DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD));
```

Comment:
Same here, static singleton `ProjectionPath` so we do not need to always decode it.